dali/pipeline/data/tensor_list.h

```diff
@@ -132,8 +132,8 @@ class DLL_PUBLIC TensorList : public Buffer<Backend> {
 
     use_copy_kernel &= (std::is_same<SrcBackend, GPUBackend>::value || other.is_pinned()) &&
                        (std::is_same<Backend, GPUBackend>::value || pinned_);
-    type.template Copy<SrcBackend, Backend>(dsts.data(), srcs.data(), sizes.data(),
-                                            nsamples, stream, use_copy_kernel);
+    type_.template Copy<SrcBackend, Backend>(dsts.data(), srcs.data(), sizes.data(),
```

Comment:
What difference does it make?

Reply:
Unlike `type_`, `type` is just an identifier. This line, however, takes place after set_type (line 114), so we can safely use the member `type_` to access its extended functionality.

Reply:
type is now an enum.